### ISSUE TITLE
extra/datadog.sh should not modify Heroku config vars

### ISSUE THREAD

**Issue by amcmanus** [DataDog/heroku-buildpack-datadog#102]
We define `DD_TAGS` as a Heroku config var, only to have the script clear it. This makes a Heroku config var unavailable for later use by other tasks such as generating deploy events and tagging them to an environment. Exporting an empty `DD_TAGS` var after you're done with it is completely unexpected and incorrect behavior for the Heroku config var paradigm. Here's the snippet of code doing the wrong thing.

```
# Convert comma delimited tags from env vars to yaml
if [ -n "$DD_TAGS" ]; then
  DD_TAGS="$(sed "s/,[ ]\?/\\\n  - /g" <<< "$DD_TAGS")"
  TAGS="$TAGS\n  - $DD_TAGS"
  # User set tags are now in YAML, clear the env var.
  export DD_TAGS=""
fi
```

**Comment by jyee** [comment]
@amcmanus I think you're right that there is likely a cleaner way to do this. However, the tags are set in the DD Agent config YAML and should be applied to any metrics collected by the Agent and sent to Datadog.

Can you tell me more about what you're doing and how you're seeing that in Datadog?

**Comment by amcmanus** [comment]
Yeah, definitely. Sorry for not including an example originally.

We set `DD_TAGS` via `heroku config:set DD_TAGS=env:production -a appname`.
This can be confirmed via `heroku config:get DD_TAGS -a appname`

You're absolutely right that the datadog agent correctly uses the tag for any metrics sent to datadog, but if you want to reference this variable within a Rails process, Rake task, or anything outside the datadog agent, it is just an empty string.

Here's an example, also showing that other config vars are not affected.

```
Loading production environment (Rails 4.2.11)
irb(main):001:0> puts ENV['DD_TAGS']

=> nil
irb(main):002:0> puts ENV['DD_LOG_LEVEL']
ERROR
=> nil
```

This came up when attempting to emit an event on every Heroku deploy, and we wanted to set the `:tags` API parameter to include the same tag as the rest of the metrics. Since it's getting set to an empty string, that didn't work. As an interim solution, we've simply set another config var to the same value as `DD_TAGS`, but that's obviously not ideal.

Let me know if you need any additional information.

**Comment by jyee** [comment]
Thanks for the additional info. I think what we ought to do is update the buildpack so it doesnt set the tags in the yaml, but we'll also need to add the heroku tags (see https://github.com/DataDog/heroku-buildpack-datadog/blob/master/extra/datadog.sh#L41) to the env var and ensure it's exported.

**Comment by arapulido** [comment]
This is now fixed in version 1.17 of the buildpack: https://github.com/DataDog/heroku-buildpack-datadog/releases/tag/1.17
